cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -362,25 +338,35 @@ func ValidateKubeconfigsForExternalCA(outDir string, cfg *kubeadmapi.InitConfigu
 		kubeadmconstants.SchedulerKubeConfigFileName,
 	}
 
+	return handleKubeConfigFiles(validateKubeConfig, outDir, cfg, kubeConfigFileNames...)
+}
+
+// handleKubeConfigFiles verify or create all the requested kubeconfig files according to the flag.
+func handleKubeConfigFiles(fn handleKubeConfig, outDir string, cfg *kubeadmapi.InitConfiguration, kubeConfigFileNames ...string) error {
+
+	// gets the KubeConfigSpecs, actualized for the current InitConfiguration
 	specs, err := getKubeConfigSpecs(cfg)
 	if err != nil {
 		return err
 	}
 
 	for _, kubeConfigFileName := range kubeConfigFileNames {
+		// retrieves the KubeConfigSpec for given kubeConfigFileName
 		spec, exists := specs[kubeConfigFileName]
 		if !exists {
-			return errors.Errorf("couldn't retrive KubeConfigSpec for %s", kubeConfigFileName)
+			return errors.Errorf("couldn't retrieve KubeConfigSpec for %s", kubeConfigFileName)
 		}
 
-		kubeconfig, err := buildKubeConfigFromSpec(spec, cfg.ClusterName)
+		// builds the KubeConfig object
+		config, err := buildKubeConfigFromSpec(spec, cfg.ClusterName)
```

Comment:
I think here can still using the previous variable `kubeconfig`, no need to change it to `config`. What do you think?

Reply:
This variable name is different in two functions that are almost repetitive. I just use one of the variable names, which is not modifiable.

Reply:
I see that the two functions are almost repetitive, just think that you maybe using the logic of `ValidateKubeconfigsForExternalCA` function, but this does not affect.

Reply:
I agree with @pytimer . Let's use the `kubeconfig` name here.
We have both `cfg` and `config` names which point to different things and this is not good.